src/chttpd/src/chttpd_db.erl

```diff
@@ -711,7 +711,8 @@ db_doc_req(#httpd{method='GET', mochi_req=MochiReq}=Req, Db, DocId) ->
 
 db_doc_req(#httpd{method='POST', user_ctx=Ctx}=Req, Db, DocId) ->
     couch_httpd:validate_referer(Req),
-    couch_doc:validate_docid(DocId),
+    DbName = couch_db:name(Db),
+    couch_doc:validate_docid(DocId, DbName),
```

Comment:
DbName, DocId seems a better ordering, as doc id is subordinate to database. The function name is also (now) misleading.

Reply:
We are still validating document IDs only. DbName is used as context for validation.

Reply:
What @nickva said.